clients/src/test/java/org/apache/kafka/common/utils/MockTime.java

```diff
@@ -19,33 +19,53 @@
  */
 public class MockTime implements Time {
 
-    private long nanos = 0;
-    private long autoTickMs = 0;
+    private final long autoTickMs;
+
+    // Values from `nanoTime` and `currentTimeMillis` are not comparable, so we store them separately to catch bugs
```

Comment:
The changes in this class make sense but not sure the comment is accurate in terms of catching bugs - don't see any tracking of incorrect sage.